esp-hal/src/lib.rs

```diff
@@ -429,6 +429,15 @@ unsafe extern "C" fn stack_chk_fail() {
     panic!("Stack corruption detected");
 }
 
+/// Provide timestamp value to esp-println as millis.
+#[doc(hidden)]
+#[no_mangle]
+pub extern "Rust" fn _esp_println_timestamp() -> u64 {
```

Comment:
This should be behind a config option (with it defaulting to `true` I guess?), so that users can provide their own timestamp mechanism if they want to.